test/tstDendrogram.cpp

```diff
@@ -99,4 +104,85 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(dendrogram_union_find, DeviceType,
   }
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(dendrogram_boruvka, DeviceType,
+                              ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  using MemorySpace = typename DeviceType::memory_space;
+
+  using ArborX::Experimental::DendrogramImplementation;
+
+  ExecutionSpace space;
+
+  // Choosing n > 5000 often results in MST producing edges of equal weight.
+  int const n = 3000;
+  auto points = ArborXTest::make_random_cloud<ArborX::Point>(space, n);
+
+  ArborX::Details::MinimumSpanningTree<MemorySpace, true> mst(space, points);
+  ArborX::Experimental::Dendrogram<MemorySpace> dendrogram(space, mst.edges);
+
+  // Because the dendrogram in the MST is permuted, we need to reorder it in the
+  // increasing edge order to compare with union-find
```

Comment:
```suggestion
  // Because the dendrogram in the MST is permuted, we need to reorder it in
  // increasing edge order to compare with union-find
```